clients/lsp-bufls.el

```diff
@@ -63,6 +64,41 @@
                   :priority 0
                   :server-id 'bufls))
 
+;; Buf CLI
+(defgroup lsp-buf nil
+  "Configuration options for buf CLI."
+  :group 'lsp-mode
+  :link '(url-lint "https://github.com/bufbuild/buf")
+  :package-version '(lsp-mode . "9.0.0"))
+
+(defcustom lsp-buf-args `("beta" "lsp")
+  "Arguments to pass to buf CLI."
+  :type '(repeat string)
+  :group 'lsp-buf
+  :package-version '(lsp-mode . "9.0.0"))
+
+(defcustom lsp-buf-path "buf"
+  "Command to run buf CLI."
+  :type 'string
+  :group 'lsp-buf
+  :package-version '(lsp-mode . "9.0.0"))
+
+(defun lsp-buf--stdio-command ()
+  "Return the command and args to start buf CLI LSP server."
+  (let ((args (list lsp-buf-path)))
+    (when (and (listp lsp-buf-args)
+               (> (length lsp-buf-args) 0))
+      (setq args (append args lsp-buf-args)))
+    args))
+
+(lsp-register-client
+ (make-lsp-client :new-connection (lsp-stdio-connection
+                                   #'lsp-buf--stdio-command)
+                  :activation-fn (lsp-activate-on "protobuf")
+                  :language-id "protobuf"
+                  :priority -1
```

Comment:
Maybe we should make this 0 and replace the deprecated bufls server? 🤔

Reply:
I tried to change priorities by https://github.com/emacs-lsp/lsp-mode/pull/4580/commits/192cc095ad811a139b6b5a638f26c37dcbb6b8d6
Does this fit your idea?

Reply:
That works! Thanks! :)